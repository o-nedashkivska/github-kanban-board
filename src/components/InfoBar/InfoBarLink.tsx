import { Button } from "antd";

interface InfoBarLinkProps {
  text: string;
  href: string;
}

const InfoBarLink: React.FC<InfoBarLinkProps> = ({
  text,
  href,
}: InfoBarLinkProps) => {
  const upCaseFirst = text[0].toUpperCase() + text.slice(1);

  return (
    <Button type="link" size="large" target="_blank" href={href}>
      {upCaseFirst}
    </Button>
  );
};

export default InfoBarLink;
