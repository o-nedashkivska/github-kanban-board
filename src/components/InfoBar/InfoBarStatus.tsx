import { Typography } from "antd";
import { BulbOutlined } from "@ant-design/icons";

import styles from "./info-bar.module.css";

const { Paragraph } = Typography;

const InfoBarStatus: React.FC = () => {
  return (
    <Paragraph className={styles["info-bar__status"]} italic>
      <BulbOutlined
        style={{ color: "#e67701" }}
        className={styles["info-bar__icon"]}
      />
      To view issues of any existing repository, please enter a valid repository
      url into the input above and click 'Load issues' button.
    </Paragraph>
  );
};

export default InfoBarStatus;
