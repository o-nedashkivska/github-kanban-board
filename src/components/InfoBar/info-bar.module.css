.info-bar__links > :first-child {
  padding-left: 0;
}
