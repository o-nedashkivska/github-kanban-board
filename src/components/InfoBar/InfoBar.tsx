import { useSelector } from "react-redux";
import { Flex, Typography } from "antd";
import { StarFilled } from "@ant-design/icons";
import InfoBarLink from "./InfoBarLink";
import InfoBarStatus from "./InfoBarStatus";

import { getCurrentRepo } from "../../store/selectors";

import styles from "./info-bar.module.css";

const { Text } = Typography;
const githubUrl = "https://github.com";

const InfoBar: React.FC = () => {
  const { name: currentRepoName, stars } = useSelector(getCurrentRepo);

  if (!currentRepoName) {
    return <InfoBarStatus />;
  }

  const [owner, repo] = currentRepoName.split("/");

  return (
    <Flex align="center" gap="large">
      <Flex align="center" className={styles["info-bar__links"]}>
        <InfoBarLink text={owner} href={`${githubUrl}/${owner}`} />
        <Text style={{ color: "#1677ff" }} strong>
          {">"}
        </Text>
        <InfoBarLink text={repo} href={`${githubUrl}/${owner}/${repo}`} />
      </Flex>
      <Text strong>
        <StarFilled
          style={{ color: "#e67701" }}
          className={styles["info-bar__icon"]}
        />
        {`${stars} K stars`}
      </Text>
    </Flex>
  );
};

export default InfoBar;
