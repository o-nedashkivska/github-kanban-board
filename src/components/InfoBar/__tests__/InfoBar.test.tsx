import { screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import { renderWithProviders } from "../../../test-utils";

import InfoBar from "../InfoBar";

describe("InfoBar", () => {
  describe("displays correct elements when status is fulfilled", () => {
    const owner = "Facebook";
    const repo = "React";

    const preloadedState = {
      currentRepo: {
        name: `${owner.toLowerCase()}/${repo.toLowerCase()}`,
        stars: 220,
        status: "fulfilled",
        statusReason: "",
      },
    };

    test("displays starIcon when status is fulfilled", () => {
      renderWithProviders(<InfoBar />, { preloadedState });

      const starIcon = screen.getByRole("img");

      expect(starIcon).toBeInTheDocument();
      expect(starIcon).toHaveAttribute("aria-label", "star");
    });

    test("displays stars number when status is fulfilled", () => {
      renderWithProviders(<InfoBar />, { preloadedState });

      expect(
        screen.getByText(`${preloadedState.currentRepo.stars} K stars`)
      ).toBeInTheDocument();
    });

    test("displays two links when status is fulfilled", () => {
      renderWithProviders(<InfoBar />, { preloadedState });

      expect(screen.getByText(owner)).toBeInTheDocument();
      expect(screen.getByText(">")).toBeInTheDocument();
      expect(screen.getByText(repo)).toBeInTheDocument();

      const listLinks = screen.getAllByRole("link");
      expect(listLinks.length).toBe(2);
    });

    test("matches snapshot", () => {
      const { asFragment } = renderWithProviders(<InfoBar />, {
        preloadedState,
      });

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("displays correct elements when status is other than fulfilled", () => {
    const preloadedState = {
      currentRepo: {
        name: "facebook/react",
        stars: 0,
        status: "rejected",
        statusReason: "Something went wrong.",
      },
    };

    test("displays status icon when status is other than fulfilled", () => {
      renderWithProviders(<InfoBar />, { preloadedState });

      const statusIcon = screen.getByRole("img");

      expect(statusIcon).toBeInTheDocument();
    });

    test("displays status reason when status is other than fulfilled", () => {
      renderWithProviders(<InfoBar />, { preloadedState });

      expect(
        screen.getByText(preloadedState.currentRepo.statusReason)
      ).toBeInTheDocument();
    });

    test("matches snapshot", () => {
      const { asFragment } = renderWithProviders(<InfoBar />, {
        preloadedState,
      });

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
