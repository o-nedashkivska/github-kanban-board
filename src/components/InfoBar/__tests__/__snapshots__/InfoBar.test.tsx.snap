// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBar displays correct elements when status is fulfilled matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-center ant-flex-gap-large"
  >
    <div
      class="info-bar__links ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-center"
    >
      <a
        class="ant-btn css-dev-only-do-not-override-1kuana8 ant-btn-link ant-btn-lg"
        href="https://github.com/facebook"
        tabindex="0"
        target="_blank"
      >
        <span>
          Facebook
        </span>
      </a>
      <span
        class="ant-typography css-dev-only-do-not-override-1kuana8"
        style="color: rgb(22, 119, 255);"
      >
        <strong>
          &gt;
        </strong>
      </span>
      <a
        class="ant-btn css-dev-only-do-not-override-1kuana8 ant-btn-link ant-btn-lg"
        href="https://github.com/facebook/react"
        tabindex="0"
        target="_blank"
      >
        <span>
          React
        </span>
      </a>
    </div>
    <span
      class="ant-typography css-dev-only-do-not-override-1kuana8"
    >
      <strong>
        <span
          aria-label="star"
          class="anticon anticon-star info-bar__icon"
          role="img"
          style="color: rgb(230, 119, 1);"
        >
          <svg
            aria-hidden="true"
            data-icon="star"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
            />
          </svg>
        </span>
        220 K stars
      </strong>
    </span>
  </div>
</DocumentFragment>
`;

exports[`InfoBar displays correct elements when status is other than fulfilled matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-typography info-bar__status css-dev-only-do-not-override-1kuana8"
  >
    <i>
      <span
        aria-label="warning"
        class="anticon anticon-warning info-bar__icon"
        role="img"
        style="color: rgb(255, 77, 79);"
      >
        <svg
          aria-hidden="true"
          data-icon="warning"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M464 720a48 48 0 1096 0 48 48 0 10-96 0zm16-304v184c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V416c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8zm475.7 440l-416-720c-6.2-10.7-16.9-16-27.7-16s-21.6 5.3-27.7 16l-416 720C56 877.4 71.4 904 96 904h832c24.6 0 40-26.6 27.7-48zm-783.5-27.9L512 239.9l339.8 588.2H172.2z"
          />
        </svg>
      </span>
      Something went wrong.
    </i>
  </div>
</DocumentFragment>
`;
